src/Options/Auto.hpp

```diff
@@ -46,33 +50,33 @@ class Auto {
   std::optional<T> value_{};
 };
 
-template <typename T, AutoLabel Label>
+template <typename T, typename Label>
 bool operator==(const Auto<T, Label>& a, const Auto<T, Label>& b) noexcept {
   return static_cast<const std::optional<T>&>(a) ==
          static_cast<const std::optional<T>&>(b);
 }
 
-template <typename T, AutoLabel Label>
+template <typename T, typename Label>
 bool operator!=(const Auto<T, Label>& a, const Auto<T, Label>& b) noexcept {
   return not(a == b);
 }
 
-template <typename T, AutoLabel Label>
+template <typename T, typename Label>
 std::ostream& operator<<(std::ostream& os, const Auto<T, Label>& x) noexcept {
   const std::optional<T>& value = x;
   if (value) {
     return os << get_output(*value);
   } else {
-    return os << Label;
+    return os << pretty_type::short_name<Label>();
   }
 }
 
-template <typename T, AutoLabel Label>
+template <typename T, typename Label>
 struct create_from_yaml<Auto<T, Label>> {
   template <typename Metavariables>
   static Auto<T, Label> create(const Option& options) {
     try {
-      if (options.parse_as<std::string>() == get_output(Label)) {
+      if (options.parse_as<std::string>() == pretty_type::short_name<Label>()) {
```

Comment:
What do you think about using something like `Options::name<Label>()` so that the defining string can be overloaded with a `name()` function or similar?